### PR TITLE
Compatible with redux-saga 1.0.0

### PR DESCRIPTION
**Pull request by jp928**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (VincentLanglet) [inline .vscode/launch.json]: IDE settings should not be in the pr

File: src/expectSaga/index.js [hunk 2]
```diff
@@ -53,13 +57,13 @@ function extractState(reducer: Reducer, initialState?: any): any {
...
 }
 
-function isHelper(fn: Function): boolean {
-  return (
-    fn === takeEveryHelper ||
-    fn === takeLatestHelper ||
-    fn === takeLeadingHelper
-  );
-}
+// function isHelper(fn: Function): boolean {
```
Comment (VincentLanglet) [inline src/expectSaga/index.js]: Why not removing all the commented code

File: .vscode/launch.json (hunk 1 above)
Comment (jp928) [reply .vscode/launch.json]: @VincentLanglet
Thanks for your review. I will revise them.

File: src/expectSaga/index.js [hunk 3]
```diff
@@ -44,7 +41,8 @@ import {
   TAKE,
 } from '../shared/keys';
 
-const { asEffect, is } = utils;
+// const { asEffect, is } = utils;
```
Comment (VincentLanglet) [inline src/expectSaga/index.js]: You forgot here

File: src/expectSaga/index.js [hunk 4]
```diff
@@ -560,7 +536,8 @@ export default function expectSaga(
   api.put.like = createEffectTester(
     'put',
     PUT,
-    effects.put,
+    // effects.put,
```
Comment (VincentLanglet) [inline src/expectSaga/index.js]: here

File: src/testSaga/index.js [hunk 5]
```diff
@@ -2,7 +2,9 @@
 import isEqual from 'lodash.isequal';
 import assign from 'object-assign';
 
-import { effects, eventChannel } from 'redux-saga';
+import * as effects from 'redux-saga/effects';
+
+// import { eventChannel } from 'redux-saga';
```
Comment (VincentLanglet) [inline src/testSaga/index.js]: and here

**Comment by VincentLanglet** [comment]
@jfairbank This PR is really needed. Thanks if you can look at

File: src/expectSaga/index.js (hunk 3 above)
Comment (jp928) [reply src/expectSaga/index.js]: My bad. Will revise.

**Comment by feugy** [comment]
Thanks a million @VincentLanglet  and @jfairbank!

@jfairbank, until this is published on NPM, would you mind building the sources and committing them in your repository?
Si we could use it as dependency in package.json files: `"redux-saga-test-plan": "jp928/redux-saga-test-plan"`

**Comment by jp928** [comment]
> Thanks a million @VincentLanglet and @jfairbank!
> 
> @jfairbank, until this is published on NPM, would you mind building the sources and committing them in your repository?
> Si we could use it as dependency in package.json files: `"redux-saga-test-plan": "jp928/redux-saga-test-plan"`

You can use git link in your package.json. It is little bit ugly but before this is merged it would be a solution for time being.

**Comment by GioSensation** [comment]
@jp928 Thanks for your work on this. I hope it is merged soon. But in the meantime, could you please explain in more details how to include your version in my current project?

I have this in my package.json:

`"redux-saga-test-plan": "git+https://github.com/jfairbank/redux-saga-test-plan.git#be0e52bfcf0b08346d52b42bd753929e1c61d06e",`

This of course got me your latest changes, but when I try running the tests, I get `Cannot find module 'redux-saga-test-plan'`.

**Comment by feugy** [comment]
@GioSensation you have to build react-saga-test-plan by yourself.
```
cd node_modules/react-sata-test-plan
npm install
npm run build
```

That's why I've asked to include the built files in the repo, to avoid this manual step...

**Comment by davidjbradshaw** [comment]
@jp928 @VincentLanglet @feugy Looks like this project is looking for a new maintainer
https://twitter.com/elpapapollo/status/1069693818985795584

**Comment by fttr** [comment]
@GioSensation I think you have to use the fork from @jp928 instead of @jfairbank:
> `"redux-saga-test-plan": "git+https://github.com/jp928/redux-saga-test-plan.git#be0e52bfcf0b08346d52b42bd753929e1c61d06e",`

**Comment by GioSensation** [comment]
@fttr The hash is the same, so it actually works. You still have to `npm run build` manually, though.

After this and a few tweaks here and there I got most tests working. The only failing tests are the ones that try to assert the use of `delay`. In the new `redux-saga` version, it is now an effect and it has been moved in `redux-saga/effects`. Anybody else experiencing this? I get `TypeError: jasmine.Spec.isPendingSpecException is not a function`. I couldn't figure it out as yet.

**Comment by jfairbank** [comment]
Hi, @jp928. Thanks so much for doing this and offering to maintain! Sorry to everyone that has been waiting on this and other fixes. Life has made open source time very difficult for me.

@jp928 Would you mind updating the CHANGELOG with any important changes before we merge?

**Comment by jp928** [comment]
> Hi, @jp928. Thanks so much for doing this and offering to maintain! Sorry to everyone that has been waiting on this and other fixes. Life has made open source time very difficult for me.
> 
> @jp928 Would you mind updating the CHANGELOG with any important changes before we merge?

Hi, @jfairbank 
Thanks for your reply. The APIs not change at all but I will update CHANGELOG anyways.
I may start to make pull requests based on the issue lists in the future.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,8 +1,9 @@
 ## v4.0.0-beta.1
```
Comment (davidjbradshaw) [inline CHANGELOG.md]: Won't this be `v4.0.0-beta.2` as `1` is already published

Comment (jp928) [reply CHANGELOG.md]: Sure. Updated.

Comment (davidjbradshaw) [reply CHANGELOG.md]: That should most likely be a new entry, with just what is in your PR listed. Rather than editing beta 1 release notes

Comment (jp928) [reply CHANGELOG.md]: @davidjbradshaw 
Gotcha.

**Comment by davidjbradshaw** [comment]
@jfairbank this looks good to go now

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "redux-saga-test-plan",
-  "version": "4.0.0-beta.2",
+  "version": "4.0.0-beta.3",
```
Comment (davidjbradshaw) [inline package.json]: Did `4.0.0-beta.2` get pushed to npm? As I don't see it.

Comment (jp928) [reply package.json]: Nope. So we should stay on beta.2?

**Comment by jfairbank** [comment]
Published to npm under the beta tag! Thanks @jp928 for the work and others for the feedback!

**Pull request merged**